prowler/providers/aws/aws_provider.py

```
@@ -1313,21 +1313,30 @@ def read_aws_regions_file() -> dict:
     return data
 
 
-def get_aws_available_regions() -> set:
+def get_aws_available_regions(partition: str = None) -> set:
     """
     Get the available AWS regions from the AWS services JSON file.
 
+    Args:
+        - partition (str): The AWS partition name. Default is None.
+
     Returns:
         set: A set of available AWS regions.
     """
     try:
         data = read_aws_regions_file()
 
         regions = set()
-        for service in data["services"].values():
-            for partition in service["regions"]:
-                for item in service["regions"][partition]:
-                    regions.add(item)
+        if partition is None:
+            for service in data["services"].values():
+                for partition in service["regions"]:
+                    for item in service["regions"][partition]:
+                        regions.add(item)
+        else:
+            for service in data["services"].values():
+                if partition in service["regions"]:
```

You can raise the new error here.

---

Done